engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -147,12 +168,30 @@ func (pool *EnclavePool) GetEnclave(
 	return enclaveInfo, nil
 }
 
+// Close stop the EnclavePool sub-routine in charge of filling the pool
+// and removes all the idle enclaves already created
+func (pool *EnclavePool) Close() error {
+
+	// will terminate running processes in the sub-routine
+	pool.cancelCtxFunc()
+
+	// closing this channel will quit the sub-routine
+	close(pool.closeChan)
```

Comment:
Where are you closing the other channels? That's also why it's an overhead to use a channel for the enclaveUuids. You need to make sure it's closed, etc.

Reply:
Added the close